perllib/FixMyStreet/App/Controller/Report/New.pm

```diff
@@ -1038,7 +1041,10 @@ sub process_user : Private {
 
     my $parsed = FixMyStreet::SMS->parse_username($params{username});
     my $type = $parsed->{type} || 'email';
-    $type = 'email' unless $c->cobrand->sms_authentication || $c->stash->{contributing_as_another_user};
+    $type = 'email'
+        unless $c->cobrand->sms_authentication
+        || $c->stash->{contributing_as_another_user}
+        || $c->stash->{ignore_logged_in_user};
```

Comment:
Aha, I think this shows the underlying issue; what you raised as point 2 in the ticket only happens if the staff user does not have the "contribute as another user" permission (which it makes sense to demand they have in order to leave reports as other users, I think). I think if you give your staff user in your new test that permission, then the existing code seems to work fine as-is without the extra `ignore_logged_in_user` changes. A staff without that permission can't leave only a phone, but can leave a different email (because any normal user can fill in the form with an email).